server/src/main/java/org/elasticsearch/search/DocValueFormat.java

```diff
@@ -236,9 +237,12 @@ private DateTime(DateFormatter formatter, ZoneId timeZone, DateFieldMapper.Resol
 
         public DateTime(StreamInput in) throws IOException {
             String formatterPattern = in.readString();
+            Locale locale = in.getTransportVersion().onOrAfter(TransportVersions.DATE_TIME_DOC_VALUES_LOCALES)
+                ? LocaleUtils.parse(in.readString())
+                : DateFieldMapper.DEFAULT_LOCALE;
             String zoneId = in.readString();
             this.timeZone = ZoneId.of(zoneId);
-            this.formatter = DateFormatter.forPattern(formatterPattern).withZone(this.timeZone);
+            this.formatter = DateFormatter.forPattern(formatterPattern).withZone(this.timeZone).withLocale(locale);
```

Comment:
Should we set this to `DEFAULT_LOCALE` rather than serialize it? It looks like you can't change it.

Reply:
It is read from the specified `DateTimeFormatter`, which can be changed by eg the new `locale` option on `RangeFieldMapper`